components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -3014,7 +3046,7 @@ def define(self):
         return CGAbstractMethod.define(self)
 
     def definition_body(self):
-        if self.descriptor.interface.isCallback():
+        if self.descriptor.interface.isCallback() or self.descriptor.interface.isNamespace():
```

Comment:
The problem is more that `GetConstructorObject` is a misnomer than this change itself. It's not a constructor either for callback interfaces.